### PR TITLE
Refactor RequestUtils.js

### PR DESCRIPTION
**Pull request by young-k**
Verified nothing broke:

image from ios 2

Verifying that I replaced all instances of renamed functions:

screen shot 2018-12-06 at 6 56 23 pm

### CODE REVIEW

File: src/utils/RequestUtils.js [hunk 1]
```diff
@@ -29,9 +29,17 @@ function createRequest(
...
-async function fetchRetry(fn: () => void, n: number = 5): any {
+/**
+ * Attempts to call [fn] [retryCount] times. Only if an exception occurs [retryCount] times, throw the exception.
+ * Else returns the result of [fn].
+ *
+ * @param {() => any} fn
+ * @param {number} retryCount
+ * @returns {any} The result of [fn]
+ */
+async function retryFetch(fn: () => any, retryCount: number = 5): any {
```
Comment (kq-li) [inline src/utils/RequestUtils.js]: prefer `fetchWithRetry`

File: src/utils/RequestUtils.js [hunk 2]
```diff
@@ -43,28 +51,39 @@ async function fetchRetry(fn: () => void, n: number = 5): any {
...
+ * @param {number} refreshInterval Ms to wait before refresh
+ * @param {number} timeout Ms to wait before request times out
+ * @param {Object} objectToUpdate
  */
-function startRequestIntervals(fn: () => void, refreshInterval: number, timeout: number, setObj: Object): void {
+function updateObjectOnInterval(
+    fn: () => void,
+    refreshInterval: number,
+    timeout: number,
+    objectToUpdate: Object,
```
Comment (kq-li) [inline src/utils/RequestUtils.js]: should be `any` since `fn` returns any

File: src/utils/RequestUtils.js [hunk 3]
```diff
@@ -43,28 +51,39 @@ async function fetchRetry(fn: () => void, n: number = 5): any {
...
+ * If [fn] completes successfully, [objectToUpdate] is set to the value of [fn()].
+ *
+ * @param {() => any} fn
+ * @param {number} refreshInterval Ms to wait before refresh
+ * @param {number} timeout Ms to wait before request times out
+ * @param {Object} objectToUpdate
  */
-function startRequestIntervals(fn: () => void, refreshInterval: number, timeout: number, setObj: Object): void {
+function updateObjectOnInterval(
+    fn: () => void,
```
Comment (kq-li) [inline src/utils/RequestUtils.js]: should be `() => any`

File: src/utils/RequestUtils.js (hunk 1 above)
Comment (young-k) [reply src/utils/RequestUtils.js]: aiya ok

File: src/utils/RequestUtils.js (hunk 2 above)
Comment (young-k) [reply src/utils/RequestUtils.js]: should be `Object` right?

**Pull request merged**